RezaBG/TUDU
# Fix/kan 22 database tests

### Fix Database Tests Setup and Add Debug Logs

**Problem:**

The database tests are passing, but the expected SQLite .tables output does not show the tables (tasks, users) after test execution. This behavior is unexpected and needs clarification.

**Approach**

**1. Updated conftest.py:**
- Added debug logs to verify tables created during test setup.
- Temporarily commented out Base.metadata.drop_all() in the setup_db teardown phase to retain the tables for manual inspection in SQLite.
- Verified table creation during test execution by adding SQL queries (SELECT name FROM sqlite_master) and logging the results.

**2. Ran tests:**
- All tests pass as expected, and debug logs confirm the tables are created during the test lifecycle.
	
**3. Observed issue:**
- Despite passing tests, manually inspecting the test.db file using SQLite (.tables) does not list the tables.

**Current Status**
- Tests pass successfully (11/11), with tasks and users created during test execution.
- Need further clarification on why tables are not visible in the SQLite file after test completion.

**Request for Guidance**
- Review the test setup and teardown logic.
- Confirm if this behavior is expected in the current implementation or suggest fixes to ensure tables persist for debugging.
	

**Steps to Reproduce**
1. Run the tests:
```bash
pytest tests
```
2. Open test.db using SQLite:
```bash
sqlite3 ./test.db
```
3. Check for tables:
```sql
.tables
```

Linked Issue

Closes #27
